rdrf/rdrf/migrations/0056_auto_20170911_1039.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.7 on 2017-09-11 10:39
+from __future__ import unicode_literals
+from django.db import migrations
+from django.db.utils import ProgrammingError
+
+def extract_model_field_data_from_clinical_data_json_field(apps, schema_editor):
+    try:
+        ClinicalData = apps.get_model("rdrf", "ClinicalData")
+        for clinical_data in ClinicalData.objects.all():
+            clinical_data.django_model = clinical_data.data.get("django_model", "")
+            clinical_data.django_id = clinical_data.data.get("django_id", 0)
+            clinical_data.context_id = clinical_data.data.get("context_id", None)
+            clinical_data.save()
+    except ProgrammingError:
+        # thrown when run on wrong db
+        print("not running clinical data update on main db")
```

Comment:
Should this fail the migration?

Reply:
We had as issue like this before - the database router is meant to route any operations on a model to the right db - but for the data migration it runs on both despite the model being listed in the router as residing on the clinical db. By catching this error that was thrown , I allow the migrations to continue where this one gets run on the correct db, rather than halt the whole process.